zebra-chain/src/transaction/serialize.rs

```diff
@@ -215,7 +215,7 @@ impl ZcashDeserialize for Option<sapling::ShieldedData<SharedAnchor>> {
         //
         // Type is `B^{[ℓ_{Sapling}_{Merkle}]}`, i.e. 32 bytes
         let shared_anchor = if spends_count > 0 {
-            Some(reader.read_32_bytes()?.into())
+            Some(reader.read_32_bytes()?.try_into()?)
```

Comment:
Optional: this might also work:

```suggestion
            Some((&mut reader).zcash_deserialize_into()?)
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/3544/commits/6a8b4b8a18f4206da6e712cf6fc31d8ee5a55c78